src/span.rs

```diff
@@ -389,13 +401,90 @@ where
         }
     }
 }
+impl<T> InjectToBinary<T> for SpanContextState
+where
+    T: Write,
+{
+    fn inject_to_binary(context: &SpanContext, carrier: &mut T) -> Result<()> {
+        let mut u64buf: [u8; 8];
+        let u32buf: [u8; 4];
+        let u8buf: [u8; 1];
+
+        u64buf = context.state().trace_id.high.to_be_bytes();
+        carrier
+            .write(&u64buf)
+            .expect("could not write trace_id.high");
```

Comment:
Please return `Error(..)` instead of panicing the current thread.
You can handle I/O errors as follows:
```rust
use crate::error::from_io_error;

...

track!(carrier.write(&u64buf).map_err(from_io_error))?;
```

Reply:
Thanks, fixed.